src/events/tab_event_handler.ts

```diff
@@ -0,0 +1,75 @@
+import yaml from "js-yaml";
+import vscode from 'vscode';
+import path from 'path';
+import fs from 'fs';
+
+import { VSCODE } from "../constants";
+import { ITab, ITabFile } from "../interface";
+import { formatDatetime, getBranch } from "../utils/common";
+import { ConfigUtils } from "../utils/config_utils";
+import { pathUtils } from "../utils/path_utils";
+import { generateSettings } from "../settings";
+import { RepoState } from "../utils/repo_state_utils";
+import { UserState } from "../utils/user_utils";
+
+export class tabEventHandler {
+	repoPath = "";
+	branch = "";
+	pathUtils: any;
+
+	// Diff props
+	settings = generateSettings();
+	shouldProceed = false;
+
+	constructor(repoPath="") {
+		const userState = new UserState();
+		const isValidAccount = userState.isValidAccount();
+		this.repoPath = repoPath || pathUtils.getRootPath();
+		if(!this.repoPath) return;
+		const repoState = new RepoState(this.repoPath).get();
+		const repoIsConnected = repoState.IS_CONNECTED;
+		this.shouldProceed = isValidAccount && repoIsConnected;
+		if (!this.shouldProceed) return;
+		this.branch = getBranch(this.repoPath);
+		this.pathUtils = new pathUtils(this.repoPath, this.branch);
+	}
+
+	handleTabChangeEvent = (isTabEvent: boolean = true) => {
+		if(!this.repoPath || !isTabEvent) return;
+		// Record timestamp
+		const created_at = formatDatetime(new Date().getTime());
+		// For the current open repoPath, get the repo_id and from config File
+		const configUtils = new ConfigUtils();
+		const repoId = configUtils.getRepoIdByPath(this.repoPath);
+		if (!repoId) return
+
+		// Get list of current tabs
+		const open_tabs = vscode.window.tabGroups.all;
+		const tabs: ITabFile[] = [] = open_tabs.flatMap(tab_group => 
+			tab_group.tabs.map(tab => {
+				// Get path of tab
+				// @ts-ignore
+				const fileRelativePath = tab.input.uri.path;
+				// Get file ID using path
+				const fileId = configUtils.getFileIdByPath(this.repoPath, this.branch, fileRelativePath);
+				return {file_id: fileId || null, path: fileRelativePath};
```

Comment:
I think we don't need to do `fileId || null` because getFileIdByPath returns either number or null.

Reply:
I had added it bec the file_id of file non-connected wasn't showing, the tabs data was looking like: 

![image](https://github.com/codesyncapp/codesync-vs/assets/15974895/16957d0a-470c-47d3-a64c-a7ea65f7aa0b)

Reply:
> I think we don't need to do `fileId || null` because getFileIdByPath returns either number or null.

`getFileIdByPath` returns null in two scenarios:
- if (!this.isConfigValid()) 
- if (!repoBranchConfig)

We get `undefined` bec code searches for `fileRelPath` in `repoBranchConfig`. 

Since our file 2 doesn't exist in `repoBranchConfig`, it returns `undefined`.

![image](https://github.com/codesyncapp/codesync-vs/assets/15974895/98cf8f85-a5c5-4051-9a5a-54f7eb5f281f)

Which is why I propose we update the return statement as:

`return repoBranchConfig[fileRelPath] || null;`